.github/workflows/php-quality-checks.yml

```diff
@@ -0,0 +1,607 @@
+      needs.prepare.outputs.has-composer == 'true' &&
+      needs.prepare.outputs.has-phpunit == 'true'
+    runs-on: ubuntu-latest
+    container:
+      image: ${{ needs.prepare.outputs.php-container }}
+    steps:
+      - uses: actions/checkout@v4
+      
+      - name: Install system dependencies
+        run: |
+          apk add --no-cache git
+          
+          # Install Xdebug for coverage
+          apk add --no-cache $PHPIZE_DEPS
+          pecl install xdebug
+          docker-php-ext-enable xdebug
+
+      - uses: actions/download-artifact@v4
+        with:
+          name: vendor-php${{ inputs.php-version }}
+
+      - name: Extract vendor directory
+        run: tar -xzf vendor.tar.gz
+
+      - name: Run PHPUnit tests
+        run: |
+          echo "🧪 Running PHPUnit tests with coverage..."
+          
+          # Check if PHPUnit is installed
+          if [ -f "vendor/bin/phpunit" ]; then
+            phpunit_cmd="vendor/bin/phpunit"
+          else
+            echo "❌ PHPUnit not found in vendor/bin/"
+            exit 1
+          fi
+
+          # Build PHPUnit command with coverage
+          cmd="php -d memory_limit=${{ inputs.memory-limit }} -d xdebug.mode=coverage $phpunit_cmd"
+          cmd="$cmd --coverage-text --coverage-clover=coverage.xml"
+          
+          # Add debug flags if enabled
+          if [ "${{ inputs.debug }}" = "true" ]; then
+            cmd="$cmd --verbose"
+          fi
+
+          # Execute PHPUnit
+          eval $cmd
+
+      - name: Check coverage threshold
+        run: |
+          threshold=${{ inputs.coverage-threshold }}
+          if [ -f "coverage.xml" ] && [ "$threshold" != "0" ]; then
+            echo "📊 Checking coverage threshold ($threshold%)..."
+            
+            # Extract coverage percentage from clover XML
+            coverage=$(php -r "
+              \$xml = simplexml_load_file('coverage.xml');
+              \$metrics = \$xml->project->metrics;
+              \$covered = (float)\$metrics['coveredstatements'];
+              \$total = (float)\$metrics['statements'];
+              \$percentage = \$total > 0 ? (\$covered / \$total) * 100 : 0;
+              echo round(\$percentage, 2);
+            ")
+            
+            echo "📊 Current coverage: $coverage%"
+            
+            # Compare with threshold
+            if (( $(echo "$coverage < $threshold" | bc -l) )); then
+              echo "❌ Coverage $coverage% is below threshold $threshold%"
+              exit 1
+            else
+              echo "✅ Coverage $coverage% meets threshold $threshold%"
+            fi
+          else
+            echo "ℹ️ Skipping coverage threshold check"
+          fi
+
+      - name: Upload coverage reports
+        uses: actions/upload-artifact@v4
+        if: always()
+        with:
+          name: coverage-php${{ inputs.php-version }}
+          path: coverage.xml
+          retention-days: 7
+
+  # Matrix testing across PHP versions
```

Comment:
Not required to be a matrix. If the provided php version passes the tests that's all that's required